### PR TITLE
Allow use of Redis over sock and not TCP Socket

### PR DESCRIPTION
**Pull request by PhilETaylor**
Pull Request for Issue #15822.

### Summary of Changes

Improve tooltip to show what values should be accepted
Allow Joomla to connect correctly to Redis that is using sock and not TCP
LANGUAGE FILE CHANGE (no idea how to propagate that to translators) 

### Testing Instructions

Set up redis.conf with port 0 and listening on a sock
`port 0`
`unixsocket /tmp/redis.sock`
restart redis

Joomla global Config as such:
screen shot 2017-05-05 at 13 56 08

### Expected result

no matter what port you enter, joomla will ignore it, as long as the sock file is correct, Joomla can connect and cache

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_config.ini [hunk 1]
```diff
@@ -149,11 +149,11 @@ COM_CONFIG_FIELD_REDIS_AUTH_DESC="Redis server authentication."
 COM_CONFIG_FIELD_REDIS_AUTH_LABEL="Redis Server Authentication"
 COM_CONFIG_FIELD_REDIS_DB_DESC="Redis database."
 COM_CONFIG_FIELD_REDIS_DB_LABEL="Redis Database"
-COM_CONFIG_FIELD_REDIS_HOST_DESC="Redis server host."
-COM_CONFIG_FIELD_REDIS_HOST_LABEL="Redis Server Host"
+COM_CONFIG_FIELD_REDIS_HOST_DESC="Redis server hostname, IP or path to sock file"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_config.ini]: Please add a closing period

File: administrator/language/en-GB/en-GB.com_config.ini [hunk 2]
```diff
@@ -149,11 +149,11 @@ COM_CONFIG_FIELD_REDIS_AUTH_DESC="Redis server authentication."
...
+COM_CONFIG_FIELD_REDIS_HOST_LABEL="Redis Server"
 COM_CONFIG_FIELD_REDIS_PERSISTENT_DESC="Persistent Redis."
 COM_CONFIG_FIELD_REDIS_PERSISTENT_LABEL="Persistent Redis"
-COM_CONFIG_FIELD_REDIS_PORT_DESC="Redis server port."
+COM_CONFIG_FIELD_REDIS_PORT_DESC="Redis server TCP port, ignored if using a sock file"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_config.ini]: Please add a closing period

**Comment by brianteeman** [comment]
> no idea how to propagate that to translators) 

Don't worry about that.

File: administrator/language/en-GB/en-GB.com_config.ini (hunk 2 above)
Comment (PhilETaylor) [reply administrator/language/en-GB/en-GB.com_config.ini]: Done

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_config.ini]: Thanks

**Comment by steelAXIS** [comment]
Tested and working as expected on my side.

**Comment by franz-wohlkoenig** [comment]
@steelAXIS can i alter your Test as successfully at Issue Tracker?

**Comment by steelAXIS** [comment]
@franz-wohlkoenig sure go ahead.

**Comment by brianteeman** [comment]
@PhilETaylor can you resolve the conflicts please?

**Comment by PhilETaylor** [comment]
@brianteeman This is not a conflict. This is 10 tonnes of Joomla progressing (libraries/joomla/cache doesnt exist any more, lots of refactoring of libraries and vendors) while a PR remains unmerged. This PR is no longer fit for use.

**Comment by mbabker** [comment]
Phil it seriously should just be a `git merge upstream staging` and done.  The vast majority of PRs touching moved libraries files have been easily sync'd with a merge, I've seen very few merge conflicts beyond the file rename.

(And I wrote this before realizing the new PR was there, but just for general reference...)

**Comment by PhilETaylor** [comment]
i was/am in bed and exhausted and too sleepy to dig :) 

You cannot merge changes to a file that no longer exists though right ? This is what happened with this PR.

**Comment by mbabker** [comment]
As long as the file was moved in a git rename operation then merging the upstream branch should track that and attempt to migrate your changes to the new file path.

You're screwed though if git tracks it as deleting the old file and creating a new file (git proper, not how GitHub displays things).

**Pull request closed without merging**